modules/server_new/src/main/scala/observe/server/keywords/DhsClientHttp.scala

```diff
@@ -238,9 +243,15 @@ private class DhsClientSim[F[_]: FlatMap: Logger](date: LocalDate, counter: Ref[
 
 object DhsClientSim {
   def apply[F[_]: Sync: Logger]: F[DhsClient[F]] =
-    (Sync[F].delay(LocalDate.now), Ref.of[F, Int](0)).mapN(new DhsClientSim[F](_, _))
+    Clock[F].monotonic
+      .map(d => Instant.EPOCH.plusNanos(d.toNanos))
+      .map(LocalDateTime.ofInstant(_, ZoneId.systemDefault))
+      .flatMap(apply)
 
-  def apply[F[_]: Sync: Logger](date: LocalDate): F[DhsClient[F]] =
-    Ref.of[F, Int](0).map(new DhsClientSim[F](date, _))
```

Comment:
This should have worked. Unless DhsClientSim was reinstantiated for every step.

Reply:
It doesn't work when you restart the server.